### PR TITLE
Judge null for key and value in attachment in RpcContextFilter.

### LINKED ISSUE #2127: NullPointExexception  in RpcContextFilter
- [ ] I have searched the [issues](https://github.com/apache/incubator-dubbo/issues) of this repository and believe that this is not a duplicate.
- [ ] I have checked the [FAQ](https://github.com/apache/incubator-dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 2.6.1 - newest

### Step to reproduce this issue

1.  
String value = null;
RpcContext.getContext().getAttachments().put("demokey", value)
2. 
then invoke one method will NullPointException

**The Exception reason:
In RpcContextFilter.java，public void filter(ClientRequestContext requestContext) throws IOException
for (Map.Entry<String, String> entry : RpcContext.getContext().getAttachments().entrySet()) {
            if (entry.getValue().contains(",") || entry.getValue().contains("=")

the entry.getValue not judge null!!**

### PR DESCRIPTION
**Pull request by carryxyh**
Judge null for key and value in attachment in RpcContextFilter.
If not, NullPointerException will be thrown in RpcContextFilter in some case.

Fix #2127 : https://github.com/apache/incubator-dubbo/issues/2127#issuecomment-408344327

### CODE REVIEW

File: dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/RpcContextFilter.java [hunk 1]
```diff
@@ -70,9 +69,17 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
...
+            String value = entry.getValue();
+            if (StringUtils.isNotEmpty(key)) {
+                if (illegalForRest(key)) {
+                    throw new IllegalArgumentException("The attachments of " + RpcContext.class.getSimpleName() + " must not contain ',' or '=' when using rest protocol");
+                }
+            }
+            if (StringUtils.isNotEmpty(value)) {
+                if (illegalForRest(value)) {
+                    throw new IllegalArgumentException("The attachments of " + RpcContext.class.getSimpleName() + " must not contain ',' or '=' when using rest protocol");
+                }
```
Comment (diecui1202) [inline dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/RpcContextFilter.java]: if ( (StringUtils.isNotEmpty(key) && illegalForRest(key)) || (StringUtils.isNotEmpty(value) && illegalForRest(value)) ) {
    throw new IllegalArgumentException("The attachments of " + RpcContext.class.getSimpleName() + " must not contain ',' or '=' when using rest protocol");
}

Is the above code is better ?

File: dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/RpcContextFilter.java [hunk 2]
```diff
@@ -85,4 +92,14 @@ public void filter(ClientRequestContext requestContext) throws IOException {
...
     }
+
+    /**
+     * If a string value illegal for rest protocol(',' and '=' is illegal for rest protocol).
+     *
+     * @param v string value
+     * @return true for illegal
+     */
+    private boolean illegalForRest(String v) {
+        return v.contains(",") || v.contains("=");
```
Comment (diecui1202) [inline dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/RpcContextFilter.java]: Or judge null place here ?

Comment (carryxyh) [reply dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/RpcContextFilter.java]: Prefer judge null in 'illegalForRest'.
The meaning of this method is that if a value is not null, it should not contains ',' or '='.
I will fix this soon.
:)

**Comment by diecui1202** [comment]
LGTM.

**Pull request merged**